rest/request.go

```diff
@@ -55,6 +55,12 @@ func (r *Request) BaseUrl() *url.URL {
 		scheme = "http"
 	}
 
+	// HTTP/2.0 gives the default scheme as HTTP even when used with TLS
+	// Check if version 2.0 and TLS is not nil and given back https scheme
+	if scheme == "http" && r.ProtoMajor >= 2 && r.TLS != nil {
```

Comment:
Is it necessary to check that ProtoMajor ? What about HTTP 1.1 over HTTPS ? 

(Thanks for the PR!)